commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -0,0 +1,171 @@
+            Qt.LeftDockWidgetArea, self.dockwidget_controlled_structures)
+        # Show active models
+        self.dockwidget_controlled_structures.combobox_input_model.addItems(
+            self.databases.keys())
+        self.update_dockwidget_ids()
+        self.dockwidget_controlled_structures.show()
+
+    def update_dockwidget_ids(self):
+        """
+        Function to update the control structures dockwidget.
+        By clicking on a different model in the GUI, the id's
+        for the measuring points and structures are updated.
+        """
+        self.dockwidget_controlled_structures.\
+            combobox_input_measuring_point_id.clear()
+        self.dockwidget_controlled_structures.\
+            combobox_input_structure_id.clear()
+        db_key = self.dockwidget_controlled_structures.combobox_input_model\
+            .currentText()  # name of database
+        db = get_database_properties(db_key)
+        control_structure = ControlledStructures(
+            flavor=db["db_entry"]['db_type'])
+        control_structure.start_sqalchemy_engine(db["db_settings"])
+        # Get all id's of the connection nodes
+        try:
+            with control_structure.engine.connect() as con:
+                rs = con.execute(
+                    '''SELECT id FROM v2_connection_nodes;'''
+                )
+                connection_node_ids = rs.fetchall()
+                for connection_node_id in connection_node_ids:
+                    id_nr = connection_node_id[0]
+                    self.dockwidget_controlled_structures.\
+                        combobox_input_measuring_point_id.addItem(str(id_nr))
+        except (OperationalError, ProgrammingError) as e:
+            if "unable to open database file" in str(e):
+                msg = "Database not found."
+            elif "no such table" in str(e):
+                msg = "Table {} not found.".format("v2_connection_nodes")
+            else:
+                msg = "".format(e)
+            messagebar_message(
+                "Error", msg, level=QgsMessageBar.CRITICAL, duration=5)
+        try:
+            with control_structure.engine.connect() as con:
+                rs = con.execute(
+                    '''SELECT weir_id FROM v2_weir_view;'''
+                )
+                weir_ids = rs.fetchall()
+                for weir_id in weir_ids:
+                    id_nr = weir_id[0]
+                    self.dockwidget_controlled_structures.\
+                        combobox_input_structure_id.addItem(str(id_nr))
+        except (OperationalError, ProgrammingError) as e:
+            if "unable to open database file" in str(e):
+                msg = "Database not found."
+            elif "no such table" in str(e):
+                msg = "Table {} not found.".format("v2_weir_view")
+            else:
+                msg = "".format(e)
+            messagebar_message(
+                "Error", msg, level=QgsMessageBar.CRITICAL, duration=5)
```

Comment:
what I meant before is: imagine there will be 5 more tables for which you want to add ids to the dockwidget, then you have to write basically the same code 5 more times. Beter to use a method, something like:

```
def add_ids_to_dockwidget(control_structure, table_name):
    try:
	    with control_structure.engine.connect() as con:
		rs = con.execute(
		    '''SELECT id FROM {};'''.format(table_name)
		)
		ids = rs.fetchall()
		for _id in ids:
		    id_nr = _id[0]
		    self.dockwidget_controlled_structures.\
		        combobox_input_measuring_point_id.addItem(str(id_nr))
	except (OperationalError, ProgrammingError) as e:
	    if "unable to open database file" in str(e):
		msg = "Database not found."
	    elif "no such table" in str(e):
		msg = "Table {} not found.".format("v2_connection_nodes")
	    else:
		msg = "".format(e)
	    messagebar_message(
		"Error", msg, level=QgsMessageBar.CRITICAL, duration=5)
```